### DISCUSSION TITLE
Possible BC break in 5.4 (from 4.4) - Container generation or event dispatcher?

### DISCUSSION THREAD

**Discussion by pkly** [symfony/symfony#49601] [Q&A]
Hello, I'm currently in the process of upgrading one of our applications.

It's one on a bigger side and I've noticed an issue regarding container class generation
In 4.4 the following setup works fine

```
Service Foo (non-lazy) containing a list of interfaces C
Service Bar contains Foo

One of the services, let's call it D implements C and contains service Bar
```

However in 5.4 the same exact setup produces a container generation such that an infinite loop is created when fetching Foo as the following occurs:

1. Foo is fetched with list of interfaces
2. D contains Bar
3. Bar contains Foo, but Foo is not yet added to the service list and as such cannot be fetched from the private list and must call the method
4. Repeat 1 to 3

This causes an infinite loop and an out of memory error soon after.

Still, it's weird that this setup worked in the previous major version...

It appears that the method of container autogeneration has changed somewhat and does not attempt to check if a recursion inclusion would occur when creating service instances.

**Comment by ahundiak** [comment]
I would suggest creating a new 5.4 skeleton project and adding minimalist examples of the services causing issues.  It is often easier to read code than to read a description.

**Reply by pkly** [reply]
I'll do some more debugging and see if I can identify the root issue

**Reply by pkly** [reply]
I have figured out what the issue was and edited the post. I'll see about making a skeleton project showcasing the bug.
